### DISCUSSION TITLE
Ability to remove a stock item from admin

### DISCUSSION THREAD

**Discussion by cedum** [solidusio/solidus#4545] [New Features or Ideas]
The product stock' admin UI allows adding and editing stock items but not deleting them for a stock location.
Do you know if there're any reasons why there is no way to delete an existing stock item for a specific stock location (by using the admin UI)? And would it make sense to introduce a 'delete' button?

We have a case where we'd like to be able to keep specific variants associated only to a subset of stock locations. It would make sense to be able to discard the stock items that we don't need. This way we can implement different stock allocation & fulfillment flows based on the products' stock locations association.

**Comment by kennyadsl** [comment]
Mhh, can't think about a real reason but, could this break existing orders with line items connected to those stock items?

**Comment by seand7565** [comment]
Deleting a stock_item that still has stock has worrying implications IMO, but I don't see any reason why you shouldn't be able to delete a stock_item with exactly 0 stock. I think the workflow of emptying out a stock_item before deleting it would make sense to a lot of users.

**Comment by modded-nate** [comment]
Wanted to jump in because I think this is a two part question/issue. First there is the issue that a stock location cannot be deleted (or unassociated to a particular product or variant) once added. We have also had issue where a return using the incorrect stock location has now created a stock location to product relationship that is incorrect and now we have a stock location associated for ever. The second issue is that deleting a product currently results in inventory units and stock locations still be present that are not destroyed when the product itself is destroyed. We see this in our inventory reports, with rows or inventory associated to deleted SKUs making the data sloppy. Also any completed orders, since the product has been deleted now have no information associated in that particular order's admin section (no photo, etc.) since it's just pulling the fresh data every time an order is accessed rather than recording or saving the item information on order completion.
